core/base/src/main/java/alluxio/underfs/UfsStatusCache.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.underfs;
+
+import alluxio.AlluxioURI;
+
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * A class to hold a set of statuses which map from AlluxioURI to UfsStatus.
+ */
+public class UfsStatusCache {
+  private final Map<AlluxioURI, UfsStatus> mStatuses;
+  private final AlluxioURI mUfsUri;
```

Comment:
Can you add a comment on what this is? Also, will this URI also be in the map above?

Reply:
I will add a comment.

`mUfsUri` is used before calling `loadMetadataInternal` to see if the cache that was populated previously corresponds to the same path.

The mapping is alluxio path -> UfsStatus

Reply:
comments added